### PR TITLE
Clicking on the magnifying glass links to logs again

### PR DESCRIPTION
**Pull request by kennyt**
px

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @jfurrow

**Comment by rcorral** [comment]
@kennyt found one small issue, the dropdown doesn't populate as it should, before it would show you in the list all of the *.log files that were in the directory that you came from. Should be an easy fix.
![gif](http://cl.ly/0A3n2N17102k/Screen%20Recording%202016-06-25%20at%2012.37%20PM.gif0

File: src/js/pages/task-details/TaskLogsTab.js [hunk 1]
```diff
@@ -98,12 +99,17 @@ class TaskLogsTab extends React.Component {
...
   handleViewChange(currentFile) {
-    this.setState({
-      currentFile,
-      watching: 0,
-      searchString: '',
-      totalFound: 0
-    });
+    let currentRoutes = this.context.router.getCurrentRoutes();
+    let lastRoute = currentRoutes[currentRoutes.length - 1];
+    console.log('in here', lastRoute);
```
Comment (MatApple) [inline src/js/pages/task-details/TaskLogsTab.js]: 🍷

File: src/js/pages/task-details/TaskDetail.js [hunk 2]
```diff
@@ -139,10 +139,18 @@ class TaskDetail extends mixin(InternalStorageMixin, TabsMixin, StoreMixin) {
...
 
   handleOpenLogClick(selectedLogFile) {
-    this.setState({selectedLogFile, currentTab: 'debug'});
+    let params = Object.assign(
+      {},
+      this.props.params,
+      {filePath: encodeURIComponent(selectedLogFile.get('path'))}
+    );
+    let currentRoutes = this.context.router.getCurrentRoutes();
+    let {logRouteName} = currentRoutes[currentRoutes.length - 1];
```
Comment (orlandohohmeier) [inline src/js/pages/task-details/TaskDetail.js]: How about `let {logRouteName} = currentRoutes.pop()` which is a little less error prone.

**Comment by kennyt** [comment]
@rcorral i've been trying to look into that problem and i just realized that it's also on master. it's a really weird problem where `TaskDetail` re-mounts when you change tabs so then it fetches the directory over again and blows away the old one.

this problem has probably been here since we made each of the tabs routed

File: src/js/pages/task-details/TaskDetail.js (hunk 2 above)
Comment (rcorral) [inline src/js/pages/task-details/TaskDetail.js]: Not sure if this will cause a mutation.

**Comment by rcorral** [comment]
@kennyt alright, let's merge this in. Could you take a look at a fix for the issue though?

File: src/js/pages/task-details/TaskLogsTab.js (hunk 1 above)
Comment (rcorral) [inline src/js/pages/task-details/TaskLogsTab.js]: Let's address this.

File: src/js/pages/task-details/TaskLogsTab.js [hunk 3]
```diff
@@ -188,6 +194,7 @@ class TaskLogsTab extends React.Component {
   }
 
   onItemSelection(obj) {
+    console.log(obj);
```
Comment (rcorral) [inline src/js/pages/task-details/TaskLogsTab.js]: And this.

File: src/js/pages/task-details/TaskDetail.js (hunk 2 above)
Comment (orlandohohmeier) [inline src/js/pages/task-details/TaskDetail.js]: Good, point. Let's keep it as it is.

**Pull request merged**